nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java

```diff
@@ -1175,6 +1175,44 @@ public void lambdaPlusRestrictiveNegative() {
         .doTest();
   }
 
+  @Test
+  public void OptionalEmptinessHandlerTest() {
+    compilationHelper
+        .setArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:UnannotatedSubPackages=com.uber.lib.unannotated"))
+        .addSourceLines(
+            "TestNegative.java",
+            "package com.uber;",
+            "import java.util.Optional;",
+            "import javax.annotation.Nullable;",
+            "public class TestNegative {",
+            "  void foo() {",
+            "    Optional<Object> a = Optional.of(null);",
+            "      // no error since a.isPresent() is called",
+            "      if(a.isPresent()){",
+            "         a.get().toString();",
+            "       }",
+            "  }",
+            "}")
+        .addSourceLines(
+            "TestPositive.java",
+            "package com.uber;",
+            "import java.util.Optional;",
+            "import javax.annotation.Nullable;",
+            "public class TestPositive {",
+            "  void foo() {",
+            "    Optional<Object> a = Optional.of(null);",
+            "    // BUG: Diagnostic contains: dereferenced expression a.get() is @Nullable",
+            "    a.get().toString();",
```

Comment:
This doesn't seem right; `Optional.of(null)` will throw a `NullPointerException` because `Optional` can't wrap `null`. So this `a.get()` won't be called, and in other contexts `Optional#get` will never yield `null`.

Or do I misunderstand completely? :thinking:

Reply:
From what I understand this does not matter. As we are not going to run this code. We just run the static analysis on this code. So as long as we initialise `a` with any method it is alright. 

Anyway if you think I can write this test differently. Let me know.

Reply:
I agree that in the context of this test no NPE will happen, but the test does show that the plugin would make a "false statement" in case it were applied to real code. After all, `Optional#get()` is never `@Nullable`.

A more realistic test could be:
```java
Optional<Object> a = Optional.empty();"
// BUG: Diagnostic contains: dereferenced expression a.get() may be empty",
a.get().toString();",
```

... but that does require customizing the error message.

Reply:
Thanks 🙂

Reply:
We should actually be modeling `Optional.of()` such that the parameter is `@NonNull`.  We should add a library model here.

Reply:
Customizing the message might not be trivial. What is the actual result of `Optional<Object> a = Optional.empty(); a.get()` ? An exception?

Reply:
> What is the actual result of `Optional<Object> a = Optional.empty(); a.get()` ? An exception?

Yep:
```
$ jshell
|  Welcome to JShell -- Version 11.0.2
|  For an introduction type: /help intro

jshell> Optional<Object> a = Optional.empty()
a ==> Optional.empty

jshell> a.get()
|  Exception java.util.NoSuchElementException: No value present
|        at Optional.get (Optional.java:148)
|        at (#2:1)
```

W.r.t. your customization suggestion: SGTM :)